src/main/java/kr/allcll/seatfinder/AuthInterceptor.java

```diff
@@ -0,0 +1,49 @@
+package kr.allcll.seatfinder;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+import org.springframework.web.servlet.ModelAndView;
+
+@Component
+public class AuthInterceptor implements HandlerInterceptor {
+
+    private static final String TOKEN_KEY = "token";
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) {
+        if (isNotExistToken(request)) {
+            ThreadLocalHolder.SHARED_TOKEN.set(TokenProvider.createToken());
+            return true;
+        }
+        ThreadLocalHolder.SHARED_TOKEN.set(findTokenAtCooke(request));
+        return true;
+    }
+
+    @Override
+    public void postHandle(HttpServletRequest request, HttpServletResponse response, Object handler,
+        ModelAndView modelAndView) {
+        if (isNotExistToken(request)) {
+            response.addCookie(new Cookie(TOKEN_KEY, ThreadLocalHolder.SHARED_TOKEN.get()));
+        }
+    }
+
+    private boolean isNotExistToken(HttpServletRequest request) {
+        if (request.getCookies() == null) {
+            return true;
+        }
+        return Arrays.stream(request.getCookies())
+            .noneMatch(cookie -> cookie.getName().equals(TOKEN_KEY));
+    }
+
+    private String findTokenAtCooke(HttpServletRequest request) {
```

Comment:
오타가 있어요! 

```suggestion
    private String findTokenAtCookie(HttpServletRequest request) {
```

그리고 at 보다 from은 어떤가요? "~로부터 찾는다"라는 느낌으로 가고 싶어서요!